:bug: Unit tests failing intermittently on CI

Despite #3884, we still get random unit test failures related to the plane stress calculation. #3027 #3856 are issues and PRs related to this issue occurring before. 

I believe due to the random nature of failures (on the CI, failure or lack of failure is deterministic for a give system) that this could be related to software versions (e.g. `libmath`) being slightly different. We should find out what software is causing this and update the skipping criteria added in #3884 to reflect the actual cause.


```
================================ tests coverage ================================
_______________ coverage: platform linux, python 3.10.19-final-0 _______________

Coverage XML written to file coverage.xml
=========================== short test summary info ============================
FAILED tests/unit/test_tfcoil.py::test_plane_stress[planestressparam2] - assert array([ 1.953...71252995e+05]) == approx([9.769...4 ± 0.171253])
  
  comparison failed. Mismatched elements: 1 / 300:
  Max absolute difference: 9.769733861957293e-08
  Max relative difference: 0.5
  Index | Obtained               | Expected                       
  0     | 1.9539467723914586e-07 | 9.769733861957293e-08 ± 1.0e-12
=========== 1 failed, 647 passed, 103 warnings in 104.20s (0:01:44) ============
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -101,7 +101,7 @@ def skip_if_incompatible_system():
     if not system_compatible():
         pytest.skip(
             "This test could fail on your system due to differences caused by "
-            "floating-point rounding error"
+            "floating-point rounding differences in np.linalg.solve"
         )
 
 

--- a/tests/system_check.py
+++ b/tests/system_check.py
@@ -11,6 +11,7 @@ can be identified and compared against a known system.
 """
 
 import numpy as np
+import pytest
 
 
 def _hilbert_matrix(n):
@@ -38,4 +39,4 @@ def system_compatible():
     rel_diff = np.linalg.norm(x - x_perturbed) / np.linalg.norm(x)
 
     # check that relative difference against a known solution
-    return rel_diff == 5.8813403984318865e-05
+    return rel_diff == pytest.approx(5.8813403984318865e-05)

